js/modules/k6/experimental/csv/csv.js

```diff
@@ -0,0 +1,25 @@
+import { open } from 'k6/experimental/fs'
+import csv from 'k6/experimental/csv'
+
+let file;
+(async function () {
+	file = await open('data.csv');
+})();
+
+export default async function() {
+	let parser = new csv.Parser(file, {
```

Comment:
For context @joanlopez and @codebien I just found out that currently, because we don't have support for await in the init context, the only way (with the current design) to instantiate the parser from the init context is the following:

```javascript
let file;
let parser;
(async function () {
	file = await open('data.csv');
	parser = new csv.Parser(file, {
		delimiter: ',',
		skipFirstLine: true,
		fromLine: 3,
		toLine: 13,
	})
})();
```

Which works, but adds a workaround to the workaround :-/

An alternative I would consider would be to have the parser support for receiving a file path, and opening the file under the hood (which might involve exposing the currently private `openImpl` function from the `fs` module to other modules such as this one). 

What do you think? 🦖

Reply:
>Which works, but adds a workaround to the workaround :-/

It doesn't sound like a problem of this scope/pull-request. The problem is async in Init context and at some point we have to resolve it (we are already on it with ESM).

Reply:
You're probably right indeed 👍🏻 🙇🏻